pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -137,6 +143,14 @@ func (p *enrichedSourceProvider) GetJSON(updated cdcevent.Row) (json.JSON, error
 	if p.opts.mvccTimestamp {
 		p.jsonNonFixedData[fieldNameMVCCTimestamp] = json.FromString(updated.MvccTimestamp.AsOfSystemTime())
 	}
+	if p.opts.updated {
+		ts := updated.MvccTimestamp
+		if updated.SchemaTS.WallTime > ts.WallTime {
+			ts = updated.SchemaTS
+		}
```

Comment:
might be worth passing in the encoder's event context for this rather than redoing this logic. see https://github.com/asg0451/cockroach/blob/721f46ea60b7ea3811ce62ef61ff8b0ced1b35fd/pkg/ccl/changefeedccl/encoder_json.go#L452

in that case, probably want to do it for mvccts as well

ditto for avro